packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -530,6 +530,15 @@ def run_ai_cell(self, args: CellArgs, prompt: str):
             provider_params["request_schema"] = args.request_schema
             provider_params["response_path"] = args.response_path
 
+            # Validate that the request schema is well-formed JSON
+            try:
+                json.loads(args.request_schema)
+            except json.JSONDecodeError as e:
+                raise ValueError(
+                    "request-schema must be valid JSON. "
+                    f"Error at line {e.lineno}, column {e.colno}: {e.msg}"
+                )
+
```

Comment:
The stack trace is hard to read, since the stack trace of both exceptions is still being printed. When raising errors inside an `except` block, it's generally preferable to use the `raise ... from None` syntax:

```suggestion
            # Validate that the request schema is well-formed JSON
            try:
                json.loads(args.request_schema)
            except json.JSONDecodeError as e:
                raise ValueError(
                    "request-schema must be valid JSON. "
                    f"Error at line {e.lineno}, column {e.colno}: {e.msg}"
                ) from None

```

This should suppress the original exception `json.JSONDecodeError` from being printed when `ValueError` is raised.

Reference: https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement